### PR TITLE
#3258 Dev Deps Upgrades

### LINKED ISSUE #3258: Dependencies upgrade
## The suggestion

Need to upgrade dependencies. I suspect it might be a good idea to separate this into multiple issues, if there are real code changes required when upgrading react-native.

**Why?**
Cause we're very behind.

## Describe alternatives you've considered
N/A

## Additional context
N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3258 - kind of, need a part 2 (maybe 3)

#### Prettier 2.2.1 https://github.com/prettier/prettier/releases
- *Breaking changes* [2.0](https://prettier.io/blog/2020/03/21/2.0.0.html)
-  Made arrow-parens default 'always'. It makes sense but is a lot of work to go through every file and update them so instead disabled within the config

#### lint-staged & Husky: 5.0.0: 
- *Breaking changes*: https://github.com/typicode/husky/releases
- Did NOT upgrade.
- Waiting for a few minors/patches as this is a significant change in API

### eslint-plugin-react
- Nothing

### eslint-plugin-prettier
- Nothing

### eslint-plugin-import
- Nothing

### eslint
- *Breaking Changes*:https://eslint.org/blog/2020/05/eslint-v7.0.0-released
- Can't find any breaking on our repo changes

### victory-native
- Nothing

### reselect
- Nothing

### currency.js
- *Breaking changes*: https://github.com/scurker/currency.js/pull/213
- Didn't upgrade - will do in own issue

### ajv
- Nothing
- New major in Beta

### moment
- Nothing

### Realm 3.1.5 -> 10 [They skipped v4,7,8,9].
- Was a bug in realm that after upgrading babel meant it wouldn't work, so upgraded in this PR.
- Fixed: Rare crash (EXC_BAD_ACCESS KERN_INVALID_ADDRESS in realm::Table::migrate_links) when a schema was updated  - Think I saw this a few times
- Collection/Objects: toJSON() method
- A lot of changes to sync and interaction wiht mongoDB, mainly
- Required to bump the node version

## Testing

- [ ] App work good?

### Related areas to think about

n/A

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-10.15.0
\ No newline at end of file
+12.20.0
```
Comment (joshxg) [inline .nvmrc:1]: bumped - up to node 15 now 🤷

Comment (wlthomson) [reply .nvmrc:1]: 🎉 .

**Review by wlthomson — APPROVED** [review]
LGTM.

**Comment by mark-prins** [comment]
I took the branch, installed and started - and it crashes on my emulator. Perhaps after the branch merging it will all come together? This is the error showing:

```
2020-12-02 17:28:11.556 5700-5745/com.msupplymobile E/SoLoader: couldn't find DSO to load: libreactnativejni.so
2020-12-02 17:28:11.558 5700-5745/com.msupplymobile E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.msupplymobile, PID: 5700
    java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libreactnativejni.so
        at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:738)
```

**Comment by joshxg** [comment]
@mark-prins 

Thanks a lot for checking it out. Maybe I should merge all the deps upgrade PRs into one? I was trying to make it easier to review, but that might be a fail.


I would like to make sure people can still build before I go breaking develop - are you able to `cd android && ./gradlew clean` as well as `yarn clean-mac-cache` ? It's also possible the java version might make a difference and I might need to bump the sdkman version

**Comment by joshxg** [comment]
What I tested with: 
image



after clearing caches using `yarn clean-mac-cache` and `cd android && ./gradlew clean`, `yarn start` works. Also deleted the repository and recloned and was OK. The `gradlew clean` step seems important - I think it is clearing the build of react-native in the cache which might account for the missing `.so`. But not 100% :)

**Comment by mark-prins** [comment]
I clean forgot about cleaning up! thanks @joshxg

**Pull request merged**